backend/sirius-web-spring-collaborative-forms/src/main/java/org/eclipse/sirius/web/spring/collaborative/forms/RepresentationsEventProcessorFactory.java

```diff
@@ -0,0 +1,106 @@
+import org.springframework.stereotype.Service;
+
+/**
+ * Used to create the representations event processors.
+ *
+ * @author gcoutable
+ */
+@Service
+public class RepresentationsEventProcessorFactory implements IRepresentationEventProcessorFactory {
+
+    private final IRepresentationsDescriptionProvider representationsDescriptionProvider;
+
+    private final IObjectService objectService;
+
+    private final List<IFormEventHandler> formEventHandlers;
+
+    private final ISubscriptionManagerFactory subscriptionManagerFactory;
+
+    private final IWidgetSubscriptionManagerFactory widgetSubscriptionManagerFactory;
+
+    private final IRepresentationRefreshPolicyRegistry representationRefreshPolicyRegistry;
+
+    public RepresentationsEventProcessorFactory(IRepresentationsDescriptionProvider representationsDescriptionProvider, IObjectService objectService, List<IFormEventHandler> formEventHandlers,
+            ISubscriptionManagerFactory subscriptionManagerFactory, IWidgetSubscriptionManagerFactory widgetSubscriptionManagerFactory,
+            IRepresentationRefreshPolicyRegistry representationRefreshPolicyRegistry) {
+        this.representationsDescriptionProvider = Objects.requireNonNull(representationsDescriptionProvider);
+        this.objectService = Objects.requireNonNull(objectService);
+        this.formEventHandlers = Objects.requireNonNull(formEventHandlers);
+        this.subscriptionManagerFactory = Objects.requireNonNull(subscriptionManagerFactory);
+        this.widgetSubscriptionManagerFactory = Objects.requireNonNull(widgetSubscriptionManagerFactory);
+        this.representationRefreshPolicyRegistry = Objects.requireNonNull(representationRefreshPolicyRegistry);
+    }
+
+    @Override
+    public <T extends IRepresentationEventProcessor> boolean canHandle(Class<T> representationEventProcessorClass, IRepresentationConfiguration configuration) {
+        return IFormEventProcessor.class.isAssignableFrom(representationEventProcessorClass) && configuration instanceof RepresentationsConfiguration;
+    }
+
+    @Override
+    public <T extends IRepresentationEventProcessor> Optional<T> createRepresentationEventProcessor(Class<T> representationEventProcessorClass, IRepresentationConfiguration configuration,
+            IEditingContext editingContext) {
+        if (IFormEventProcessor.class.isAssignableFrom(representationEventProcessorClass) && configuration instanceof RepresentationsConfiguration) {
+            RepresentationsConfiguration representationsConfiguration = (RepresentationsConfiguration) configuration;
+
+            FormDescription formDescription = this.representationsDescriptionProvider.getRepresentationsDescription();
+            Optional<Object> optionalObject = this.objectService.getObject(editingContext, representationsConfiguration.getObjectId());
+            if (optionalObject.isPresent()) {
+                Object object = optionalObject.get();
+                if (formDescription != null) {
```

Comment:
Why test for `null` if the API does not return an optional? No need for this test, `null` is not an acceptable value.

Reply:
Done